lib/IDE/CodeCompletion.cpp

```diff
@@ -2490,7 +2430,7 @@ class CompletionLookup final : public swift::VisibleDeclConsumer {
       }
     }
 
-    return eraseArchetypes(T, GenericSig);
+    return T;
```

Comment:
I'm curious: how did you ensure that every place that needs to `eraseArchetypes` in `addTypeAnnotation` has the correct generic signature passed in, especially since the generic signature is optional for that method?  Did you make it non-optional and check every call, or was there some other approach?

Reply:
I just checked all usages of `getTypeOfMember()` and passed `DC->getGenericSignatureOfContext()` to type printing functions (`addCallArgumentPatterns()` and `addTypeAnnotation()`).